controllers/configuration_controller_test.go

```diff
@@ -201,6 +202,159 @@ func TestInitTFConfigurationMetaWithJobNodeSelector(t *testing.T) {
 	assert.Equal(t, meta.JobNodeSelector, map[string]string{"ssd": "true"})
 }
 
+func TestPrepareTFVariables(t *testing.T) {
+	prjId := "PrjID"
+	prjIdValue := "test123"
+	testKey := "Test"
+	testValue := "abc123"
+	credentialKey := "key"
+	credentialValue := "testkey"
+	variable, _ := json.Marshal(map[string]interface{}{
+		testKey: testValue,
+	})
+	configuration := v1beta2.Configuration{
+		ObjectMeta: v1.ObjectMeta{
+			Namespace: "default",
+			Name:      "abc",
+		},
+		Spec: v1beta2.ConfigurationSpec{
+			ProviderReference: &crossplane.Reference{
+				Name:      "default",
+				Namespace: "default",
+			},
+			HCL: "test",
+			Variable: &runtime.RawExtension{
+				Raw: variable,
+			},
+		},
+	}
+	meta := &TFConfigurationMeta{
+		JobEnv: map[string]interface{}{
+			prjId: prjIdValue,
+		},
+		ProviderReference: &crossplane.Reference{
+			Name:      "default",
+			Namespace: "default",
+		},
+		Credentials: map[string]string{
+			credentialKey: credentialValue,
+		},
+	}
+	err := meta.prepareTFVariables(&configuration)
+	assert.Nil(t, err)
+	rTestKey := fmt.Sprintf("TF_VAR_%s", testKey)
+	wantVarSecretData := map[string]string{prjId: prjIdValue, rTestKey: testValue, credentialKey: credentialValue}
+	for k, v := range wantVarSecretData {
+		actualV, ok := meta.VariableSecretData[k]
+		assert.Equal(t, ok, true)
+		assert.Equal(t, actualV, []byte(v))
+	}
+	existMap := map[string]bool{
+		prjId:         false,
+		rTestKey:      false,
+		credentialKey: false,
+	}
+	for _, e := range meta.Envs {
+		if e.Name == prjId {
+			existMap[prjId] = true
+		}
+		if e.Name == rTestKey {
+			existMap[rTestKey] = true
+		}
+		if e.Name == credentialKey {
+			existMap[credentialKey] = true
+		}
+	}
+	for _, v := range existMap {
+		assert.Equal(t, v, true)
+	}
```

Comment:
This can't test the case that we don't put anything in meta.Envs. Let's change it to:
```go
switch e.Name:
case prjID:
	existMap[prjID] = true
(...omitted...)
default:
	t.Fatalf("unexpected %s, e.Name)

assert.Equal(len(existMap),3)
for _, v := range existMap {
	assert.Equal(t, v, true)
}
```